### PR TITLE
OEUI-30: UI implementation (React Components) of the Order Entry page

### PR DESCRIPTION
**Pull request by zeze111**
## JIRA TICKET NAME:

[OEUI-30: UI implementation (React Components) of the Order Entry page](https://issues.openmrs.org/browse/OEUI-30)

### SUMMARY:

Designed and implemented the order entry in react, created components for the each section of the order UI page. These components consists of the proper fields and designs to show patient's details, type of patient (out/in), search and add new order, view all active orders and view all past orders. The add new order includes a form that allows a user choose from the standard dose prescriptions or free text prescription. The standard dose consists of: the dose to take, the frequency for the dose to be taken, the route to be used, the reason for the order, the total quantity of the dose, the date to start the order, how long the dose should be taken for, and a field for additional information.

### CODE REVIEW

File: app/js/components/orderEntry/ActiveOrders.jsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                        </tbody>
+                      </table>
+                    </td>
+                    <td className="text-center">
+                      <a href="#"> <i className="fas fa-edit"></i> </a>
+                      <a href="#"> <i className="fas fa-times"></i> </a>
+                    </td>
+                  </tr>
+                </tbody>
+              </table >
```
Comment (fuchodeveloper) [inline app/js/components/orderEntry/ActiveOrders.jsx]: Could you check for the tags causing the red highlighting?

File: app/js/components/orderEntry/ActiveOrders.jsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                      started 12/03/2018
+                    </td>
+                    <td className="text-center" style={{ fontSize: "14px" }}>
+                      Active
+                    </td>
+                    <td style={{ paddingTop: "0" }}>
+                      <table className="table" style={{ marginBottom: "0" }}>
+                        <tbody>
+                          <tr style={{ fontSize: "14px" }}>
+                            <td> Oral </td>
```
Comment (dkayiwa) [inline app/js/components/orderEntry/ActiveOrders.jsx]: Did you intentionally hard code these?

**Comment by dkayiwa** [comment]
This is such a huge pul request. Did you just combine every body's work in this PR?

File: package.json [hunk 3]
```diff
@@ -69,7 +75,8 @@
     "build:prod": "npm run test && npm run build",
     "build:deploy": "webpack --progress --colors --mode=deploy --target=web",
     "watch": "webpack --progress --colors --watch --mode=deploy --target=web",
-    "test": "node_modules/.bin/mocha tests/**/*.test.* --require tests/setup.js",
+    "est": "node_modules/.bin/mocha tests/**/*.test.* --require tests/setup.js",
```
Comment (Betty-Kebenei) [inline package.json]: I had not seen this, you deleted a 't' in 'test'. Please readd it.

File: app/js/components/orderEntry/ActiveOrders.jsx (hunk 2 above)
Comment (zeze111) [reply app/js/components/orderEntry/ActiveOrders.jsx]: Yes I did, the components are just skeletons for the team to build on, so we can easily consume the Api and have consistent UI

File: package.json (hunk 3 above)
Comment (zeze111) [reply package.json]: that was intentional, the actual 'test' is jest, i think if you scroll down some more, you would see it

**Comment by zeze111** [comment]
@dkayiwa no, the changes are all mine, I created components and wrote tests for them

**Comment by dkayiwa** [comment]
@zeze111 are the data rows that i see, hard coded? For instance the active and past orders.

**Comment by zeze111** [comment]
@dkayiwa yes they are, like I said, it's just a skeleton, so when we consume the Api, the data rows will be made to read from the database

**Pull request merged**